### PR TITLE
Migrate SamplePoint Latitude and Longitude fields to Location (GPSCoordinates)

### PR DESCRIPTION
**Pull request by xispa**
## Description of the issue/feature this PR addresses

This Pull Request migrates the Latitude and Longitude fields from SamplePoint (`CoordinateField`) to another field named `Location`, that is a `GPSCoordinatesField`.

## Current behavior before PR

Latitude and longitude coordinates are stored in independent fields in DMS format

## Desired behavior after PR is merged

Latitude and longitude coordinates are stored in a single field in decimal degrees format

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: src/senaite/core/content/samplepoint.py [hunk 1]
```diff
@@ -68,28 +68,17 @@ class ISamplePointSchema(model.Schema):
...
-        ),
-        description=_(
-            u"description_samplepoint_longitude",
-            default=u"Enter the Sample Point's longitude in degrees 0-180, "
-                    u"minutes 0-59, seconds 0-59.9999 and E/W indicator"
+            "description_samplepoint_location",
+            default="The geographical point or area from which samples are "
+                    "taken for further analysis. It plays a pivotal role in "
+                    "research as it can significantly influence the results "
+                    "and conclusions drawn from the study."
```
Comment (ramonski) [inline src/senaite/core/content/samplepoint.py]: We have used everywhere unicodes inside the message factories, but I can not find out if this is by purpose or by habit. At least the `zope.i18nmessageid.message.Message` inherits from `six.text_type`, which is using this differentiation:

```python
PY2 = sys.version_info[0] == 2
PY3 = sys.version_info[0] == 3
PY34 = sys.version_info[0:2] >= (3, 4)

if PY3:
    string_types = str,
    integer_types = int,
    class_types = type,
    text_type = str
    binary_type = bytes

    MAXSIZE = sys.maxsize
else:
    string_types = basestring,
    integer_types = (int, long)
    class_types = (type, types.ClassType)
    text_type = unicode
    binary_type = str
```

However, I do not grasp at the moment if this is something we need to take care of...

Maybe only if we use specific characters like `<` or `>` etc. in the descriptions?

Then I would rather stick to have it everywhere the same.

Comment (xispa) [reply src/senaite/core/content/samplepoint.py]: > We have used everywhere unicodes inside the message factories, but I can not find out if this is by purpose or by habit. 

Look at this:

```python
>>> from bika.lims import senaiteMessageFactory as _
>>> _("Something weirdº")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/senaite/buildout-cache/eggs/zope.i18nmessageid-5.1.1-py2.7-linux-x86_64.egg/zope/i18nmessageid/message.py", line 113, in __call__
    msgid_plural, default_plural, number)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 15: ordinal not in range(128)
>>> _(u"Something weirdº")
u'Something weird\xba'
```

Comment (xispa) [reply src/senaite/core/content/samplepoint.py]: So I guess we should return back to `unicode`

Comment (xispa) [reply src/senaite/core/content/samplepoint.py]: Returned back to `unicode` with https://github.com/senaite/senaite.core/pull/2566/commits/4caa168412f77edc2735e701ecfcb18f28b98f7e

**Review by ramonski — APPROVED** [review]

**Pull request merged**